src/librustc_traits/chalk/lowering.rs

```diff
@@ -448,8 +458,11 @@ impl<'tcx> LowerInto<'tcx, Option<chalk_ir::QuantifiedWhereClause<RustInterner<'
         self,
         interner: &RustInterner<'tcx>,
     ) -> Option<chalk_ir::QuantifiedWhereClause<RustInterner<'tcx>>> {
-        match &self.kind() {
-            ty::PredicateKind::Trait(predicate, _) => {
+        // FIXME(chalk): forall
+        match self.ignore_qualifiers(interner.tcx).skip_binder().kind() {
+            ty::PredicateKind::ForAll(_) => bug!("unexpected predicate: {:?}", self),
+            &ty::PredicateKind::Trait(predicate, _) => {
+                let predicate = &ty::Binder::bind(predicate);
```

Comment:
And here.

Reply:
I only rebind the binder skipped in line 462 here, the shift to get from `Trait` to `Binder(Trait)` happens inside of `ignore_qualifiers`.